SIS_import/sis_upload.rb

```diff
@@ -154,69 +249,92 @@ def parseJson(data)
 p "archive directory: " + archiveDirectory
 p "output directory: " + outputDirectory
 
-if (Dir[tokenFile].length != 1)
-	## token file
-        uploadError = "Cannot find token file #{tokenFile}."
-else
-        File.open(tokenFile, 'r') do |tFile|
-	        while line = tFile.gets
-	          # only read the first line, which is the token value
-	                token=line.strip
-	                break
-	        end
-        end
-	if (token=="")
-		uploadError="Empty token for Canvas upload."
-	end
-end
-
-if (uploadError)
-elsif (Dir[currentDirectory].length != 1)
-	## working directory
-	uploadError = "Cannot find current working directory " + currentDirectory
-elsif (Dir[archiveDirectory].length != 1)
-	## archive directory
-	uploadError = "Cannot find archive directory " + archiveDirectory
-elsif (Dir[outputDirectory].length != 1)
+if (Dir[outputDirectory].length != 1)
 	## logs directory
-	uploadError = "Cannot find logs directory " + outputDirectory
-else
-	# the canvas import zip file
-	fileNames = Dir[currentDirectory+ "Canvas_Extract_*.zip"];
-	if (fileNames.length == 0)
-		## cannot find zip file to upload
-		uploadError = "Cannot find SIS zip file"
-	elsif (fileNames.length > 1)
-		## there are more than one zip file
-		uploadError = "There are more than one SIS zip files to be uploaded."
-	elsif
-		## get the name of file to process
-		fileName=fileNames[0]
-		currentFileBaseName = File.basename(fileName)
-
-		# upload start time
-		p "upload start time : " + Time.new.inspect
-
-		# upload the file to canvas server
-		uploadError = upload_to_canvas(fileName, token, server, outputDirectory)
-
-
-		# upload stop time
-		p "upload stop time : " + Time.new.inspect
-	end
-end
+	upload_error = "Cannot find logs directory " + outputDirectory
 
-if (!uploadError)
-	## if there is no upload error
-	# move file to archive directory after processing
-	FileUtils.mv(fileName, archiveDirectory+currentFileBaseName)
-	p "SIS upload finished with " + fileName
-	exit
-else
 	## check first about the environment variable setting for MAILTO '
-	p " use the environment variable for sending out error messages #{ENV['MAILTO']}"
+	p "Use the environment variable 'MAILTO' for sending out error messages to #{ENV['MAILTO']}"
 	## send email to support team with the error message
-	`echo #{uploadError} | mail -s "#{server} Upload Error" #{ENV['MAILTO']}`
-	abort(uploadError)
+	`echo #{upload_error} | mail -s "#{server} Upload Error" #{ENV['MAILTO']}`
```

Comment:
Just as Ruby offers HTTP clients, it also offers SMTP clients.  Replace this external call to mail with a function call to a SMTP function/method.